### ISSUE TITLE
Meta-ticket: Migration from patchbot to GitHub Actions

### ISSUE THREAD

**Issue by tobiasdiez** [sagemath/sage#33457]
<div id="comment:0"></div>

Most of the CI infrastructure runs already on Github actions. For a full migration from the patchbot to github actions, the following checks need to be migrated:
- Build and run doctests: Done.
- commit_messages: Not needed. Git doesn't allow empty messages.
- coverage: replaced by codecov #33355
- deprecation_number: if the code contains a deprecation warning, the ticket number in the deprecation should match the actual ticket number. **Not yet implemented in Github Actions.**
- python3: looks for some patterns in a python or rst file.
    - `xrange` (not valid in python3)
    - `.iterkeys`, `.itervalues`, `.iteritems` (not valid in python3)
    - `basestring` (not valid in python3)
    - `__nonzero__`  (not valid in python3)
  No need to implement in Github Actions.
- python3_pyx: looks for some patterns in a cython file
    - `import six` and `from six import` 
  Done via relint #30467
- pyflakes: replaced by pyright #33456, refs #30415
- pycodestyle: Done
- blocks: checks for the blocks:
    - correct syntax is `.. SEEALSO::` (done via relint #30467)
    - `TESTS` and `EXAMPLES` should be plural, `NOTE` singular (done via relint #30467)
    - no `::` after `INPUT` and `OUTPUT` blocks (done via relint #30467)
    - no `::` after `REFERENCE` blocks (done via relint #30467)
    - no ` :` at the end of lines (seems discarded in #30467)
    - no `Returns` at the start of lines (seems discarded in #30467)
- triple_colon: looks for the presence of triple colons `:::` or `: ::`. Done via relint #30467.
- trac_links: looks for the presence of badly formatted trac roles `:trac:`, missing the initial colon. Done via relint #30467.
- startup_time: measures the time to startup sage (the time to the prompt). **Not yet implemented in Github Actions.**
- startup_modules: the number of modules imported at startup. **Not yet implemented in Github Actions.**
- docbuild: Done
- git_rev_list: No need to implement in Github Actions.

Above list should be merged into https://trac.sagemath.org/wiki/patchbot/plugins

CC:  @mkoeppe @fchapoton

Component: **build**

_Issue created by migration from https://trac.sagemath.org/ticket/33457_

**Comment by tobiasdiez** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,7 +1,7 @@
 Most of the CI infrastructure runs already on Github actions. For a full migration from the patchbot to github actions, the following checks need to be migrated:
 - Build and run doctests: Done
 - commit_messages: Still needed? Git doesn't allow empty messages (except if explicitly overwritten with the flag allow-empty-message)
-- coverage
+- coverage: replaced by codecov #33355
 - deprecation_number
 - python3: Done (via relint, right?)
 - python3_pyx: Done (via relint, right?)
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -14,3 +14,6 @@
 - startup_modules
 - docbuild: Done
 - git_rev_list: Still needed?
+
+Above list should be merged into https://trac.sagemath.org/wiki/patchbot/plugins
+
``````

**Comment by kwankyu** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,6 +1,6 @@
 Most of the CI infrastructure runs already on Github actions. For a full migration from the patchbot to github actions, the following checks need to be migrated:
 - Build and run doctests: Done
-- commit_messages: Still needed? Git doesn't allow empty messages (except if explicitly overwritten with the flag allow-empty-message)
+- commit_messages: Not needed. Git doesn't allow empty messages.
 - coverage: replaced by codecov #33355
 - deprecation_number
 - python3: Done (via relint, right?)
``````

**Comment by kwankyu** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -2,7 +2,7 @@
 - Build and run doctests: Done
 - commit_messages: Not needed. Git doesn't allow empty messages.
 - coverage: replaced by codecov #33355
-- deprecation_number
+- deprecation_number: if the code contains a deprecation warning, the ticket number in the deprecation should match the actual ticket number. Not yet implemented in Github Actions.
 - python3: Done (via relint, right?)
 - python3_pyx: Done (via relint, right?)
 - pyflakes: replaced by pyright #33456, refs #30415
``````

**Comment by kwankyu** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -10,8 +10,8 @@
 - blocks: Done (via relint, right?)
 - triple_colon: Done (via relint, right?)
 - trac_links: Done (via relint, right?)
-- startup_time
-- startup_modules
+- startup_time: measures the time to startup sage (the time to the prompt). Not yet implemented in Github Actions.
+- startup_modules: the number of modules imported at startup. Not yet implemented in Github Actions.
 - docbuild: Done
 - git_rev_list: Still needed?
 
``````

**Comment by kwankyu** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -13,7 +13,7 @@
 - startup_time: measures the time to startup sage (the time to the prompt). Not yet implemented in Github Actions.
 - startup_modules: the number of modules imported at startup. Not yet implemented in Github Actions.
 - docbuild: Done
-- git_rev_list: Still needed?
+- git_rev_list: No need to be implemented in Github Actions.
 
 Above list should be merged into https://trac.sagemath.org/wiki/patchbot/plugins
 
``````

**Comment by kwankyu** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -3,13 +3,28 @@
 - commit_messages: Not needed. Git doesn't allow empty messages.
 - coverage: replaced by codecov #33355
 - deprecation_number: if the code contains a deprecation warning, the ticket number in the deprecation should match the actual ticket number. Not yet implemented in Github Actions.
-- python3: Done (via relint, right?)
-- python3_pyx: Done (via relint, right?)
+- python3: looks for some patterns in a python or rst file.
+    - `xrange`
+    - `.iterkeys`, `.itervalues`, `.iteritems`
+    - `basestring`
+    - `__nonzero__`  
+  Done via relint?
+
+- python3_pyx: looks for some patterns in a cython file
+    - `import six` and `from six import` 
+  Done via relint?
 - pyflakes: replaced by pyright #33456, refs #30415
 - pycodestyle: Done
-- blocks: Done (via relint, right?)
-- triple_colon: Done (via relint, right?)
-- trac_links: Done (via relint, right?)
+- blocks: checks for the blocks:
+    - correct syntax is `.. SEEALSO::`
+    - `TESTS` and `EXAMPLES` should be plural, `NOTE` singular
+    - no `::` after `INPUT` and `OUTPUT` blocks
+    - no `::` after `REFERENCE` blocks
+    - no ` :` at the end of lines
+    - no `Returns` at the start of lines 
+  Done via relint?
+- triple_colon: looks for the presence of triple colons `:::` or `: ::`. Done via relint?
+- trac_links: looks for the presence of badly formatted trac roles `:trac:`, missing the initial colon. Done via relint?
 - startup_time: measures the time to startup sage (the time to the prompt). Not yet implemented in Github Actions.
 - startup_modules: the number of modules imported at startup. Not yet implemented in Github Actions.
 - docbuild: Done
``````

**Comment by kwankyu** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -4,27 +4,26 @@
 - coverage: replaced by codecov #33355
 - deprecation_number: if the code contains a deprecation warning, the ticket number in the deprecation should match the actual ticket number. Not yet implemented in Github Actions.
 - python3: looks for some patterns in a python or rst file.
-    - `xrange`
-    - `.iterkeys`, `.itervalues`, `.iteritems`
-    - `basestring`
-    - `__nonzero__`  
-  Done via relint?
+    - `xrange` (not valid in python3)
+    - `.iterkeys`, `.itervalues`, `.iteritems` (not valid in python3)
+    - `basestring` (not valid in python3)
+    - `__nonzero__`  (not valid in python3)
+  No need to implement in Github Actions
 
 - python3_pyx: looks for some patterns in a cython file
     - `import six` and `from six import` 
-  Done via relint?
+  Done via relint #30467
 - pyflakes: replaced by pyright #33456, refs #30415
 - pycodestyle: Done
 - blocks: checks for the blocks:
-    - correct syntax is `.. SEEALSO::`
-    - `TESTS` and `EXAMPLES` should be plural, `NOTE` singular
-    - no `::` after `INPUT` and `OUTPUT` blocks
-    - no `::` after `REFERENCE` blocks
-    - no ` :` at the end of lines
+    - correct syntax is `.. SEEALSO::` (done via relint #30467)
+    - `TESTS` and `EXAMPLES` should be plural, `NOTE` singular (done via relint #30467)
+    - no `::` after `INPUT` and `OUTPUT` blocks (done via relint #30467)
+    - no `::` after `REFERENCE` blocks (done via relint #30467)
+    - no ` :` at the end of lines 
     - no `Returns` at the start of lines 
-  Done via relint?
-- triple_colon: looks for the presence of triple colons `:::` or `: ::`. Done via relint?
-- trac_links: looks for the presence of badly formatted trac roles `:trac:`, missing the initial colon. Done via relint?
+- triple_colon: looks for the presence of triple colons `:::` or `: ::`. Done via relint #30467
+- trac_links: looks for the presence of badly formatted trac roles `:trac:`, missing the initial colon. Done via relint #30467
 - startup_time: measures the time to startup sage (the time to the prompt). Not yet implemented in Github Actions.
 - startup_modules: the number of modules imported at startup. Not yet implemented in Github Actions.
 - docbuild: Done
``````

**Comment by kwankyu** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -9,7 +9,6 @@
     - `basestring` (not valid in python3)
     - `__nonzero__`  (not valid in python3)
   No need to implement in Github Actions
-
 - python3_pyx: looks for some patterns in a cython file
     - `import six` and `from six import` 
   Done via relint #30467
``````

**Comment by kwankyu** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -19,8 +19,8 @@
     - `TESTS` and `EXAMPLES` should be plural, `NOTE` singular (done via relint #30467)
     - no `::` after `INPUT` and `OUTPUT` blocks (done via relint #30467)
     - no `::` after `REFERENCE` blocks (done via relint #30467)
-    - no ` :` at the end of lines 
-    - no `Returns` at the start of lines 
+    - no ` :` at the end of lines (seems discarded in #30467)
+    - no `Returns` at the start of lines (seems discarded in #30467)
 - triple_colon: looks for the presence of triple colons `:::` or `: ::`. Done via relint #30467
 - trac_links: looks for the presence of badly formatted trac roles `:trac:`, missing the initial colon. Done via relint #30467
 - startup_time: measures the time to startup sage (the time to the prompt). Not yet implemented in Github Actions.
``````

**Comment by kwankyu** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -8,7 +8,7 @@
     - `.iterkeys`, `.itervalues`, `.iteritems` (not valid in python3)
     - `basestring` (not valid in python3)
     - `__nonzero__`  (not valid in python3)
-  No need to implement in Github Actions
+  No need to implement in Github Actions.
 - python3_pyx: looks for some patterns in a cython file
     - `import six` and `from six import` 
   Done via relint #30467
@@ -21,12 +21,12 @@
     - no `::` after `REFERENCE` blocks (done via relint #30467)
     - no ` :` at the end of lines (seems discarded in #30467)
     - no `Returns` at the start of lines (seems discarded in #30467)
-- triple_colon: looks for the presence of triple colons `:::` or `: ::`. Done via relint #30467
-- trac_links: looks for the presence of badly formatted trac roles `:trac:`, missing the initial colon. Done via relint #30467
+- triple_colon: looks for the presence of triple colons `:::` or `: ::`. Done via relint #30467.
+- trac_links: looks for the presence of badly formatted trac roles `:trac:`, missing the initial colon. Done via relint #30467.
 - startup_time: measures the time to startup sage (the time to the prompt). Not yet implemented in Github Actions.
 - startup_modules: the number of modules imported at startup. Not yet implemented in Github Actions.
 - docbuild: Done
-- git_rev_list: No need to be implemented in Github Actions.
+- git_rev_list: No need to implement in Github Actions.
 
 Above list should be merged into https://trac.sagemath.org/wiki/patchbot/plugins
 
``````

**Comment by kwankyu** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,8 +1,8 @@
 Most of the CI infrastructure runs already on Github actions. For a full migration from the patchbot to github actions, the following checks need to be migrated:
-- Build and run doctests: Done
+- Build and run doctests: Done.
 - commit_messages: Not needed. Git doesn't allow empty messages.
 - coverage: replaced by codecov #33355
-- deprecation_number: if the code contains a deprecation warning, the ticket number in the deprecation should match the actual ticket number. Not yet implemented in Github Actions.
+- deprecation_number: if the code contains a deprecation warning, the ticket number in the deprecation should match the actual ticket number. **Not yet implemented in Github Actions.**
 - python3: looks for some patterns in a python or rst file.
     - `xrange` (not valid in python3)
     - `.iterkeys`, `.itervalues`, `.iteritems` (not valid in python3)
@@ -23,8 +23,8 @@
     - no `Returns` at the start of lines (seems discarded in #30467)
 - triple_colon: looks for the presence of triple colons `:::` or `: ::`. Done via relint #30467.
 - trac_links: looks for the presence of badly formatted trac roles `:trac:`, missing the initial colon. Done via relint #30467.
-- startup_time: measures the time to startup sage (the time to the prompt). Not yet implemented in Github Actions.
-- startup_modules: the number of modules imported at startup. Not yet implemented in Github Actions.
+- startup_time: measures the time to startup sage (the time to the prompt). **Not yet implemented in Github Actions.**
+- startup_modules: the number of modules imported at startup. **Not yet implemented in Github Actions.**
 - docbuild: Done
 - git_rev_list: No need to implement in Github Actions.
 
``````
